src/Sylius/Component/User/Model/User.php

```diff
@@ -405,29 +402,42 @@ public function serialize(): string
             $this->enabled,
             $this->id,
             $this->encoderName,
-        ]);
+        ];
     }
 
     /**
-     * @param string $serialized
+     * The serialized data have to contain the fields used by the equals method and the username.
+     *
+     * @internal
+     */
+    public function serialize(): string
+    {
+        return serialize($this->__serialize());
+    }
+
+    public function __unserialize(array $data): void
+    {
+        $this->password = $data['password'] ?? null;
+        $this->salt = $data['salt'] ?? null;
+        $this->usernameCanonical = $data['usernameCanonical'] ?? null;
+        $this->username = $data['username'] ?? null;
+        $this->locked = $data['locked'] ?? null;
+        $this->enabled = $data['enabled'] ?? null;
+        $this->id = $data['id'] ?? null;
+        $this->encoderName = $data['encoderName'] ?? null;
+    }
+
+    /**
+     * @internal
      */
     public function unserialize($serialized): void
     {
-        $data = unserialize($serialized);
+        $data = (array) unserialize((string) $serialized);
```

Comment:
```suggestion
        $data = unserialize($data);
        
        if (!is_array($data)) {
            throw new UnexpectedValueException(sprintf('Cannot deserialize %s instance', self::class));
        }
```
Should fix the Static checks.